### PR TITLE
 Fix handling of crashkernel parameter for Xen

### PR DESCRIPTION
**Pull request by imobach**
It fixes [bsc#962008](https://bugzilla.suse.com/show_bug.cgi?id=962008). The same as #58 but against SP4.

### CODE REVIEW

**Comment by lslezak** [comment]
LGTM, thanks! :+1:

File: src/Kdump.ycp [hunk 1]
```diff
@@ -981,15 +1006,12 @@ boolean WriteKdumpBootParameter () {
...
             result = Bootloader::setKernelParam (actual_boot_section, "crashkernel", crash_value);
-	} else {
-	    BootCommon::sections[section_pos, "xen_append"]= "crashkernel="+convertCrashkernelForXEN(crash_value);
-	    // added flag which means that section was changed bnc #432651
-	    BootCommon::sections[section_pos, "__changed"] = true;
-	    result = true;
-	    y2milestone("Write boot section to XEN boot section %1", BootCommon::sections[section_pos]:nil);
+	} else { // boot section is Xen
+        UpdateXenAppendParameter("crashkernel", convertCrashkernelForXEN(crash_value));
+        result = true
```
Comment (lslezak) [inline src/Kdump.ycp]: NP: `result` is set by default to `true`, see above. It seems you could even remove this line...

File: src/Kdump.ycp [hunk 2]
```diff
@@ -1013,7 +1035,13 @@ boolean WriteKdumpBootParameter () {
   } else {
     if (crashkernel_param) {
        //delete crashkernel paramter from bootloader
-       result = Bootloader::setKernelParam (actual_boot_section, "crashkernel", "false");
+       if (!is_xen()) // boot section is NOT Xen
+       {
+          result = Bootloader::setKernelParam (actual_boot_section, "crashkernel", "false");
+       } else { // boot section is Xen
+          UpdateXenAppendParameter("crashkernel", "false");
+          result = true
```
Comment (lslezak) [inline src/Kdump.ycp]: NP: The same here...

File: src/Kdump.ycp (hunk 1 above)
Comment (imobach) [inline src/Kdump.ycp]: Well, you're right. I hate long functions (I usually skip that kind of details) :)

**Comment by lslezak** [comment]
OK, LGTM<sup>2</sup>.

I'll stop looking at it to avoid finding some more cosmetic issues... :wink:

**Comment by imobach** [comment]
Thanks!

**Pull request merged**